modules/core/src/test/java/org/apache/synapse/message/store/impl/jms/JmsMessageStoreTest.java

```diff
@@ -0,0 +1,130 @@
+        SynapseConfiguration synapseConfiguration = new SynapseConfiguration();
+        synapseConfiguration.addMessageStore("JMSStore", jmsStore);
+        SynapseEnvironment synapseEnvironment = new Axis2SynapseEnvironment(synapseConfiguration);
+        Axis2MessageContext axis2MessageContext = new Axis2MessageContext(new org.apache.axis2.context.MessageContext(), synapseConfiguration, synapseEnvironment);
+        MessageContext messageContext = axis2MessageContext;
+        SOAP11Factory factory = new SOAP11Factory();
+        SOAPEnvelope envelope = factory.createSOAPEnvelope();
+        messageContext.setEnvelope(envelope);
+
+        jmsStore.init(synapseEnvironment);
+
+        JmsProducer jmsProducer = (JmsProducer) jmsStore.getProducer();
+        boolean response = jmsProducer.storeMessage(messageContext);
+
+        Assert.assertTrue("Message not stored!", response);
+    }
+
+    /**
+     * Testing whether jms consumer is consuming the message from the store
+     * @throws Exception
+     */
+    @Test
+    public void testConsumer() throws Exception{
+
+        JmsStore jmsStore = new JmsStore();
+
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("java.naming.factory.initial", "org.apache.activemq.jndi.ActiveMQInitialContextFactory");
+        parameters.put("java.naming.provider.url", "tcp://127.0.0.1:61616");
+        jmsStore.setParameters(parameters);
+        jmsStore.setName("TestJmsStore2");
+        SynapseConfiguration synapseConfiguration = new SynapseConfiguration();
+        synapseConfiguration.addMessageStore("JMSStore", jmsStore);
+        ConfigurationContext cfgCtx = new ConfigurationContext(synapseConfiguration.getAxisConfiguration());
+        SynapseEnvironment synapseEnvironment = new Axis2SynapseEnvironment(cfgCtx, synapseConfiguration);
+        Axis2MessageContext axis2MessageContext = new Axis2MessageContext(new org.apache.axis2.context.MessageContext(), synapseConfiguration, synapseEnvironment);
+        MessageContext messageContext = axis2MessageContext;
+        SOAP11Factory factory = new SOAP11Factory();
+        SOAPEnvelope envelope = factory.getDefaultEnvelope();
+        OMElement element = AXIOMUtil.stringToOM("<name><value>Test</value></name>");
+        envelope.getBody().addChild(element);
+        messageContext.setEnvelope(envelope);
+
+        jmsStore.init(synapseEnvironment);
+
+        JmsProducer jmsProducer = (JmsProducer) jmsStore.getProducer();
+        jmsProducer.storeMessage(messageContext);
+
+        JmsConsumer jmsConsumer = (JmsConsumer) jmsStore.getConsumer();
+        MessageContext responseContext = jmsConsumer.receive();
+
+        Assert.assertNotNull("Could not consume message from the store!", responseContext);
+    }
+
+    @AfterClass
+    public static void stopBroker() throws Exception{
+        if(broker.isStarted()){
+            broker.stop();
+        }
+        System.out.println("Broker Stopped!");
```

Comment:
Shall we log instead of System.out.println

Reply:
implemented a different logic